plugin-server/src/worker/vm/upgrades/historical-export/export-historical-events.ts

```diff
@@ -47,9 +47,34 @@ export function addHistoricalEventsExportCapability(
         // the historical export to duplicate them
         meta.global.timestampBoundariesForTeam = timestampBoundaries
 
+        await meta.utils.cursor.init('batch_id')
+
+        const storedTimestampCursor = await meta.storage.get(TIMESTAMP_CURSOR_KEY, null)
+        if (storedTimestampCursor) {
+            await meta.jobs.restartExportIfNeeded({ storedTimestampCursor }).runIn(10, 'minutes')
+        }
+
         await oldSetupPlugin?.()
     }
 
+    tasks.job['restartExportIfNeeded'] = {
+        name: 'restartExportIfNeeded',
+        type: PluginTaskType.Job,
+        exec: async (payload) => {
+            const storedTimestampCursor = await meta.storage.get(TIMESTAMP_CURSOR_KEY, null)
+
+            // if the cursor hasn't been incremented within 10 minutes of the restart
+            // that means we didn't pick up from where we left off automatically
+            // thus, kick off a new export chain with a new batchId
+            if (payload && payload.storedTimestampCursor === storedTimestampCursor) {
+                const batchId = await meta.utils.cursor.increment('batch_id')
```

Comment:
That won't do it. This is definitely a bit complicated.

`EXPORT_RUNNING_KEY` is only toggled off when we're **done** exporting. If the server restarts, that will still be `true`, irrespective of if we're running or not. We don't want a restart to allow a user to trigger another export, because a job might still have lingered in the queue. And as you can see, a job triggers another job, meaning one job lingering in the queue triggers the whole process.

This is handling for the job not being in the queue, which is independent of `EXPORT_RUNNING_KEY`